packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -187,6 +187,14 @@ class RunCommand extends RunCommandBase {
         hide: true,
         help: 'Whether to automatically invoke webOnlyInitializePlatform.',
       )
+      ..addFlag('fast-start',
+        negatable: true,
+        defaultsTo: false,
+        hide: true,
+        help: 'Whether to quickly bootstrap applications with a minimal app. '
+              'Currently this is only supported on Android devices. This option '
```

Comment:
Should we do a `throwToolExit` if this is passed for a non-Android device?

Reply:
It's not technically harmful - the app will perform an unnecessary hot restart but otherwise function fine

Reply:
We could either exit, exit + warn, or ignore it

Reply:
I'm thinking we could warn and force it to off. I'd like to avoid requiring the IDE plugins to implement logic for determining if the device supports it, and let the tool fallback to the old behavior automatically. That will make it easier to roll out support for new devices

Reply:
> I'm thinking we could warn and force it to off.

sgtm

Reply:
Added code to force it off, printStatus warning for non machine modes, and test case